alchimia/engine.py

```
@@ -1,26 +1,58 @@
 from __future__ import absolute_import, division
 
 from sqlalchemy.engine.base import Engine
+from sqlalchemy.exc import StatementError
 
-from twisted.internet.threads import deferToThreadPool
+from twisted._threads import ThreadWorker, AlreadyQuit
+
+from twisted.internet.defer import Deferred, fail
+from twisted.python.failure import Failure
+
+from threading import Thread
+
+try:
+    from Queue import Queue
+except:
```

Do you really want a bare except here?